gui/src-tauri/src/main.rs

```diff
@@ -286,3 +300,15 @@ fn valid_java_paths() -> Vec<(OsString, i32)> {
 		})
 		.collect()
 }
+
+fn executable<W: AsRef<OsStr> + ?Sized>(bin: &W) -> OsString {
```

Comment:
There is no comment on this function. What does it do? How about naming it `maybe_exe` and say "appends .exe on windows".

Better yet, why is this even a function? Its not like its doing anything particularly reusable. Just have 
```rust
const JAVA_EXE: &OsStr = if cfg!(windows) {
  "java.exe"
} else {
  "java"
};
```

Reply:
didnt know if would be needed for smth else

Reply:
dont add until you know you need it, when the alternative is so much simpler. You can always refactor into a new function